runtime/src/commonMain/kotlin/pbandk/FieldDescriptor.kt

```diff
@@ -125,4 +132,52 @@ class FieldDescriptor<M : Message, T>(
             override fun isDefaultValue(value: Any?) = (value as? kotlin.collections.Map<*, *>)?.isEmpty() == true
         }
     }
+
+    @Suppress("UNCHECKED_CAST")
+    fun <M : Message, T> parseUnknownField(
+        unknownField: UnknownField.Value,
+        fieldDescriptor: FieldDescriptor<M, T>
```

Comment:
It's kind of weird that this is a method on `FieldDescriptor` but it also takes a `FieldDescriptor` argument. I think it'd make more sense to make this method an extension method anyways (I'm torn whether it should be an extension on `UnknownField.Value` that takes a `FieldDescriptor` parameter, or an extension on `FieldDescriptor` that takes an `UnknownField.Value` parameter).

Reply:
I've moved it to `Unknown.Value` extension. Thanks for the suggestion, it's better for sure.